From f5ae74ca7e2b4b8bc2ffb747a3ccb70389f2390c Mon Sep 17 00:00:00 2001
From: Nikolaus Schulz <nikolaus.schulz@avionic-design.de>
Date: Fri, 21 Oct 2016 13:27:37 +0200
Subject: [PATCH] Allow using system libraries and headers

The libeigen header files, the opengl header file and libglfw3 are not
distributed with the visionworks-tracking-samples, but they're referenced by
the build. (The missing files are contained in the visionworks-samples
package, where they're used for the build as well.)

In the makefile, put the corresponding cflags and ldlfags into separate
variables, so that they can be overridden when make is invoked.

Signed-off-by: Nikolaus Schulz <nikolaus.schulz@avionic-design.de>
---
 .../visionworks-tracking/sources/samples/object_tracker/Makefile | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/usr/share/visionworks-tracking/sources/samples/object_tracker/Makefile b/usr/share/visionworks-tracking/sources/samples/object_tracker/Makefile
index 93d62df..d42367e 100644
--- a/usr/share/visionworks-tracking/sources/samples/object_tracker/Makefile
+++ b/usr/share/visionworks-tracking/sources/samples/object_tracker/Makefile
@@ -83,9 +83,14 @@ EIGEN_CFLAGS := -I../../3rdparty/eigen
 NVXIO_CFLAGS := -I../../nvxio/include -DUSE_GSTREAMER=1 -DUSE_GSTREAMER_OMX=1
 NVXIO_LIBS := ../../libs/$(OS_ARCH)/$(OSLOWER)/$(TARGET)$(if $(abi),/$(abi))/libnvxio.a
 
+OPENGL_CFLAGS := -I../../3rdparty/opengl
+
+GLFW3_CFLAGS := -I../../3rdparty/glfw3/include
+GLFW3_LIBS := ../../3rdparty/glfw3/libs/libglfw3.a
+
 INCLUDES :=
 INCLUDES += $(NVXIO_CFLAGS)
-INCLUDES +=  -I../../3rdparty/opengl -I../../3rdparty/glfw3/include  $(shell pkg-config --cflags freetype2)
+INCLUDES +=  $(OPENGL_CFLAGS) $(GLFW3_CFLAGS) $(shell pkg-config --cflags freetype2)
 INCLUDES += $(EIGEN_CFLAGS)
 INCLUDES += $(EXTERNAL_CFLAGS)
 
@@ -104,7 +109,7 @@ ifneq ($(PKG_CONFIG_SYSROOT_DIR),)
 endif
 
 LIBRARIES += $(NVXIO_LIBS)
-LIBRARIES += $(shell pkg-config --libs freetype2) ../../3rdparty/glfw3/libs/libglfw3.a $(shell pkg-config --libs gl) $(shell pkg-config --libs xrandr xi xxf86vm x11)
+LIBRARIES += $(shell pkg-config --libs freetype2) $(GLFW3_LIBS) $(shell pkg-config --libs gl) $(shell pkg-config --libs xrandr xi xxf86vm x11)
 LIBRARIES +=  $(shell pkg-config --libs gstreamer-base-1.0 gstreamer-pbutils-1.0 gstreamer-app-1.0)
 LIBRARIES +=   -lvisionworks_tracking
 LIBRARIES += $(EXTERNAL_LIBS)
-- 
2.1.4

